src/UglyToad.PdfPig/DocumentLayoutAnalysis/TextEdgesExtractor.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.Geometry;
+
+namespace UglyToad.PdfPig.DocumentLayoutAnalysis
+{
+    /// <summary>
+    /// Text edges extractor. Text edges are where words have either there BoundingBox's left, right or mid coordinates aligned on the same vertical line.
+    /// <para>Useful to detect text columns, tables, justified text, lists, etc.</para>
+    /// </summary>
+    public class TextEdgesExtractor
```

Comment:
Probably worth making this a `public static class` so that consumers don't try and new it up and get confused where the methods are, just reduces the discovery surface area of the public API.